public_html/wp-content/plugins/wordcamp-organizer-survey/includes/debrief-survey/email.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Creates the Organizer Debrief Survey email in the Camptix email queue.
+ */
+
+namespace WordCamp\OrganizerSurvey\DebriefSurvey\Email;
+
+defined( 'WPINC' ) || die();
+
+/**
+ * Constants.
+ */
+const EMAIL_KEY_ID = 'organizer_debrief_survey_email';
+
+/**
+ * Return the email ID.
+ *
+ * @return mixed
+ */
+function get_email_id() {
+	return get_option( EMAIL_KEY_ID );
+}
+
+/**
+ * Return lead organizer email.
+ *
+ *  @return string
+ */
+function get_lead_organizer_email() {
+	$meta = get_wordcamp_post()->meta;
+	return $meta['Email Address'][0];
+}
+
+/**
+ * Return lead organizer full name.
+ *
+ * @return string
+ */
+function get_lead_organizer_full_name() {
+	$meta = get_wordcamp_post()->meta;
+	return $meta['Organizer Name'][0];
+}
+
+/**
+ * Return WordCamp location.
+ *
+ * @return string
+ */
+function get_wordcamp_location() {
+	$meta = get_wordcamp_post()->meta;
+	return $meta['Location'][0];
+}
+
+
+/**
+ * Return a token to put into the survey URL.
+ *
+ * @return string
+ */
+function get_encryption_token() {
+	$wordcamp_id = get_current_blog_id();
+	return hash_hmac( 'sha1', $wordcamp_id, ORGANIZER_SURVEY_ACCESS_TOKEN_KEY );
```

Comment:
```suggestion
	return hash_hmac( 'sha256', $wordcamp_id, ORGANIZER_SURVEY_ACCESS_TOKEN_KEY );
```

I should have double checked if `sha1` was still considered secure before suggesting it; turns out it's not. `sha256` should be fine though.

Similar to `hash_equals()`, it's not really critical for this application, but a good habit to get into.

The other `sha1` instance will need to be updated to match.

Reply:
Actually I did consider the fact that `sha1` can be compromised initially. I just thought later on that maybe there was a reason you suggested `sha1`, like perhaps it doesn't need such a high level of security for this particular use case here, and `sha1` performs better in terms of performance 😆